source/postcard-rpc/src/server/mod.rs

```diff
@@ -49,6 +49,9 @@ pub trait WireTx {
     /// use your own custom type that implements [`AsWireTxErrorKind`].
     type Error: AsWireTxErrorKind;
 
+    /// Wait for the connection to be established
+    async fn wait_connection(&self) {}
```

Comment:
Would it make sense for the default impl to do a `yield_now()` from the `futures` crate or `embassy-futures` crate, so at least we don't totally live-lock the system?

Same for WireRx.

Reply:
I've gone a bit back and forward on this; This default implementation is  correct for wire protocols that don't have a concept of a connection in the first place. E.g. for serial i'd expect the Wire{Tx,Rx}  to just wrap an open serial port so no need to implement this.

For non-connection type wires a yield would just bounce back needlessly to the executor; For connection types, it will avoid the live-lock, but you'd still end up in a spinning which is not necessarily better. For now i've simply added that connection oriented protocols should implement it in the documentation :). Maybe on an next API break the default implementation(s) could be dropped instead forcing authors to make the choice?